src/doc/tarpl/other-reprs.md

```diff
@@ -0,0 +1,71 @@
+% Alternative representations
+
+Rust allows you to specify alternative data layout strategies from the default.
+
+
+
+
+# repr(C)
+
+This is the most important `repr`. It has fairly simple intent: do what C does.
+The order, size, and alignment of fields is exactly what you would expect from
+C or C++. Any type you expect to pass through an FFI boundary should have `repr(C)`,
+as C is the lingua-franca of the programming world. This is also necessary
+to soundly do more elaborate tricks with data layout such as reintepretting values
+as a different type.
+
+However, the interaction with Rust's more exotic data layout features must be kept
+in mind. Due to its dual purpose as "for FFI" and "for layout control", `repr(C)`
+can be applied to types that will be nonsensical or problematic if passed through
+the FFI boundary.
+
+* ZSTs are still zero-sized, even though this is not a standard behaviour
+  in C, and is explicitly contrary to the behaviour of an empty type in C++, which
+  still consumes a byte of space.
+
+* DSTs, tuples, and tagged unions are not a concept in C and as such are never
+  FFI safe.
+
+* **The [drop flag][] will still be added**
+
+* This is equivalent to one of `repr(u*)` (see the next section) for enums. The
+  chosen size is the default enum size for the target platform's C ABI. Note that
+  enum representation in C is undefined, and this may be incorrect when the C
+  code is compiled with certain flags.
```

Comment:
This is a tiny bit confusing -- it's talking about something that seems to be defined, but then says it isn't. More clarifications on what one should expect would be nice.